AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt

```diff
@@ -5,30 +5,112 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
 import androidx.navigation.fragment.findNavController
-import com.swtug.anticovid.view.BaseFragment
 import com.swtug.anticovid.R
+import com.swtug.anticovid.models.User
+import com.swtug.anticovid.repositories.FirebaseListener
+import com.swtug.anticovid.repositories.FirebaseRepo
 import com.swtug.anticovid.repositories.PreferencesRepo
+import com.swtug.anticovid.view.BaseFragment
 
-class ProfileFragment: BaseFragment() {
+class ProfileFragment : BaseFragment() {
 
     private lateinit var btnlogout: Button
+    private lateinit var btnedit: Button
+    private lateinit var editTextName: EditText
+    private lateinit var editTextSurname: EditText
+    private lateinit var editTextEMail: EditText
+    private lateinit var editTextAddress: EditText
+    private lateinit var editTextSocialSecurityID: EditText
+    private lateinit var editTextPhoneNumber: EditText
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
         return layoutInflater.inflate(R.layout.fragment_profile, null)
     }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         initFields(view)
+        initListeners()
+        assignValues()
+
+    }
+
+    private fun assignValues() {
+        val user = PreferencesRepo.getUser(requireContext())
+        if (user != null) {
+            editTextName.setText(user.name)
+            editTextSurname.setText(user.surname)
+            editTextEMail.setText(user.email)
+            editTextAddress.setText(user.address)
+            editTextSocialSecurityID.setText(user.secid)
+            editTextPhoneNumber.setText(user.phonenumber)
+        }
+
+    }
 
+    private fun initListeners() {
         btnlogout.setOnClickListener {
             PreferencesRepo.deleteUser(requireContext())
             PreferencesRepo.deleteVaccination(requireContext())
             findNavController().navigate(R.id.action_profileFragment_to_loginFragment)
+
+        }
+
+        btnedit.setOnClickListener {
+            val oldUser = PreferencesRepo.getUser(requireContext())
+            if (oldUser != null) {
+                val user = User(
+                    oldUser.id,
+                    editTextName.text.toString(),
+                    editTextSurname.text.toString(),
+                    oldUser.email,
+                    editTextAddress.text.toString(),
+                    editTextSocialSecurityID.text.toString(),
+                    editTextPhoneNumber.text.toString(),
+                    oldUser.password
+                )
```

Comment:
Please check some of the input fields that they aren't empty (e.g. see Register Fragment)

Reply:
Fixed with our new commit

Reply:
LGTM 👍